### PR TITLE
Add `@types/*` & declare TS modules for woocommerce-admin

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of https://github.com/woocommerce/woocommerce/issues/32581

Add `@types/testing-library__jest-dom` & declare as TS modules for non-TS modules to fix admin TS errors:

- Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement | null>'.
- Could not find a declaration file for module '@wordpress/plugins'.
- Could not find a declaration file for module '@woocommerce/tracks'.
- Cannot find module './logo.png' or its corresponding type declarations.

### How to test the changes in this Pull Request:

1. Run `pnpm install & pnpm nx build woocommerce-admin`
2. Run `pnpm nx ts:check woocommerce-admin`
3. Should not see above TS errors

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/typings/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
 declare module '@woocommerce/e2e-utils';
 declare module '@woocommerce/e2e-environment';
 declare module '@woocommerce/settings';
+declare module '@woocommerce/tracks';
 declare module '@wordpress/data';
 declare module '@wordpress/compose';
+declare module '@wordpress/plugins';
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/typings/index.d.ts]: Hmm doing this just means that we will ignore any type definitions for everything imported from @wordpress/plugins

I think a better way is to install the typings from DefinitelyTyped : https://www.npmjs.com/package/@types/wordpress__plugins

And we should do it for the @wp/data, compose too.

For @woocommerce/tracks its one of our own packages and there aren't DefinitelyTyped definitions for it. I wonder if we should accept the error and create an issue to make type definitions for it?

File: plugins/woocommerce-admin/client/typings/index.d.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 declare module 'gridicons/dist/*' {
 	const value: React.ElementType< {
 		size?: 12 | 18 | 24 | 36 | 48 | 54 | 72;
 		onClick?: ( event: MouseEvent | KeyboardEvent ) => void;
 	} >;
 	export default value;
 }
+declare module '*.png';
+declare module '*.jpg';
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/typings/index.d.ts:13]:
```suggestion
declare module '*.jpg';
declare module '*.svg';
```

Comment (rjchow) [reply plugins/woocommerce-admin/client/typings/index.d.ts:13]: I noticed there was a declaration error for .svg too 😄

**Review by rjchow — CHANGES_REQUESTED** [review]

File: plugins/woocommerce-admin/client/typings/index.d.ts (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/typings/index.d.ts]: Good catch! I didn't notice that they have `@types` packages.

> For @woocommerce/tracks its one of our own packages and there aren't DefinitelyTyped definitions for it. I wonder if we should accept the error and create an issue to make type definitions for it?

Yes, I think we could create an issue for this, but I'd like to ignore this error. Is that good?

File: plugins/woocommerce-admin/client/typings/index.d.ts (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/typings/index.d.ts:13]: Thanks, updated. 2bf5c8f5abffd36ea910db3e44a5e08f61bbb075

File: plugins/woocommerce-admin/client/typings/index.d.ts (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/typings/index.d.ts]: Added @wp/data, compose, plugin type definitions in 8df6178d3b5126d9d6461aeacc0dd2dc054a54e2

**Review by rjchow — APPROVED** [review]
Looks great! Thanks for making the changes 😃

**Pull request merged**